### ISSUE TITLE
Reading chronicle-queues of "fieldless_binary" wire type fails in some cases

### ISSUE THREAD

**Issue by bjarteaeide** [OpenHFT/Chronicle-Queue#1232]
When reading enums of certain lengths from a "fieldless_binary" queue, the method-reader fails on subsequent messages with the following error:
```
java.lang.IllegalStateException: Failed to read method name or ID
	at net.openhft.chronicle.queue.FieldlessTest$EntityListenerMethodReader.readOneCall(FieldlessTest$EntityListenerMethodReader.java:49)
	at net.openhft.chronicle.wire.AbstractGeneratedMethodReader.readOne0(AbstractGeneratedMethodReader.java:126)
	at net.openhft.chronicle.wire.AbstractGeneratedMethodReader.readOne(AbstractGeneratedMethodReader.java:233)
```

I have attached a junit test demonstrating the issue. [FieldlessTest.txt](https://github.com/OpenHFT/Chronicle-Queue/files/9879747/FieldlessTest.txt)

The test fails for both chronicle-queue versions 2.23.128 and 5.24ea0.
When using chronicle-queue version 4.16.4 it runs successfully.

**Comment by alamar** [comment]
Last version where this test runs without errors is `5.23ea23`

**Comment by alamar** [comment]
```
commit a078d1aea7e424c7e2bbac33e3964d460fded22e (HEAD)
Author: Peter Lawrey <peter.lawrey@chronicle.software>
Date:   Thu Jul 14 14:41:16 2022 +0100

    More robust reading of the dataFormat

```
candidate for a root cause

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.24ea2](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.24ea2), [BOM-2.24ea9](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.24ea9)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.23.37](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.23.37), [BOM-2.23.135](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.23.135)
